### ISSUE TITLE
cannot load prawn/table

### ISSUE THREAD

**Issue by roscom** [prawnpdf/prawn-table#37]
**/home/user/.rbenv/versions/2.1.4/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:135:in `require': cannot load such file -- prawn/table (LoadError)**

Relevant Gemfile dependices are:

``` javascript
pdf-core (0.4.0)
prawn (1.3.0)
     pdf-core (~> 0.4.0)
ttfunk (~> 1.4.0)
prawn-table (0.2.1)
prawn_rails (0.0.11)
    prawn (>= 0.11.1)
    railties (>= 3.0.0)
```

Rails 4.1.8
Ruby 2.1.4p285

Relevant includes in the pdf generator are:

``` javascript
require 'prawn'
require 'prawn/measurement_extensions'
# require 'prawn/layout'
require "prawn/table"
include ActionView::Helpers::NumberHelper
```

And it fails to find prawn/table

Help!!  Everything has now stopped with this issue.

**Comment by tnt** [comment]
I have a similar issue without Bundler. Strangely it works when I do the `require`s in the opposite order:

    [1] pry(main)> require 'prawn'
    => true
    [2] pry(main)> require 'prawn/table'
    LoadError: cannot load such file -- prawn/table
    from /home/thelonius/.rbenv/versions/2.1.5/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:128:in `require'
    [3] pry(main)> quit
    $ pry
    [1] pry(main)> require 'prawn/table'
    => true
    [2] pry(main)> require 'prawn'
    => true

And I *need* to `require 'prawn'` after `prawn-table`.  The versions are prawn 1.3.0, prawn-table 0.2.1 and Ruby 2.1.5 (same with 2.2.0).

**Comment by sandal** [comment]
@tnt, Have you checked to ensure that the correct gems are being activated? For example, if you do `require "prawn"` only, what does `Prawn::VERSION` read?

**Comment by tnt** [comment]
[1] pry(main)> require 'prawn'
    => true
    [2] pry(main)> Prawn::VERSION
    => "1.3.0"

**Comment by tnt** [comment]
After uninstalling older prawn versions (1.0.0 and 1.2.1) it works now.

**Comment by sandal** [comment]
@tnt: You don't necessarily need to uninstall them, but you do need to handle gem locking yourself if you don't want to rely on Bundler. For example, this should work even with many versions of Prawn installed:

```ruby
gem "prawn", "~> 1.3.0"
require "prawn"
require "prawn/table"
```

The failures you were seeing before were due to the fact that Prawn itself shipped with a `lib/prawn/table.rb` file.

**Comment by tnt** [comment]
Thank you, this actually almost works.

```
[1] pry(main)> gem 'prawn', "1.3.0"
=> true
[2] pry(main)> require 'prawn'
=> true
[3] pry(main)> require 'prawn/table'
LoadError: cannot load such file -- prawn/table
from /home/thelonius/.rbenv/versions/2.1.5/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:128:in `require'
[4] pry(main)> quit
$ pry
[1] pry(main)> gem 'prawn', "1.3.0"
=> true
[2] pry(main)> gem 'prawn-table', "0.2.1"
=> true
[3] pry(main)> require 'prawn'
=> true
[4] pry(main)> require 'prawn/table'
=> true
```

**Comment by sandal** [comment]
That makes sense, older versions of `prawn-table` had strict version locking, and we now have looser constraints. Really this is the problem that Bundler is meant to solve, and it can be tedious to go with out it because it means doing explicit locking as you've had to do here for each gem.

**Comment by tnt** [comment]
I will use Bundler from now on. I should have done this before; that would have saved us some time.

Thank you again and by the way thank you for `prawn`, `prawn-table` and everything. Nice to know you back again actively developing `prawn`.

**Comment by KaraAJC** [comment]
I'm having this same issue with prawn/layout/grid, is this related?

**Comment by raquelhortab** [comment]
> This isn't a bug. You need to `require "prawn"` before you `require "prawn/table"`. Make sure that if you're using Bundler, you do something like `Bundler.require` to let it handle loading the dependencies for you.

But the original question already required prawn before prawn/table.

I am having the same problem, I can't seem to find prawn/table. I got prawn 1.3.0 installed (rails 3.2, ruby 1.9.3), no other version is installed.  `require "prawn/table"`results in this error: `LoadError: cannot load such file -- prawn/table`. I've already tried requiring prawn before this. If I navigate to the gem folder I see no table subdirectory or file.
